flag.go

```diff
@@ -46,40 +56,59 @@ type FlagLoader struct {
 	// that will used in passed into the flag for Usage.
 	FlagUsageFunc func(name string) string
 
+	// ErrorHandling matches flag.ErrorHandling value
+	//  By Default this is set to flag.ExitOnError for backwards compatibility.
+	//  Other values include flag.ContinueOnError and flag.PanicOnError
+	ErrorHandling ErrorHandling
+
 	// only exists for testing.  This is the raw flagset that is to parse
 	flagSet *flag.FlagSet
 }
 
+func (f *FlagLoader) Help() string {
+	out := new(bytes.Buffer)
+	f.flagSet.SetOutput(out)
+	f.flagSet.PrintDefaults()
+	return out.String()
+}
+
 // Load loads the source into the config defined by struct s
 func (f *FlagLoader) Load(s interface{}) error {
 	strct := structs.New(s)
 	structName := strct.Name()
 
-	flagSet := flag.NewFlagSet(structName, flag.ExitOnError)
+	var flagSet *flag.FlagSet
+	switch f.ErrorHandling {
+	case ContinueOnError:
+		flagSet = flag.NewFlagSet(structName, flag.ContinueOnError)
+		// need to prevent default printing of flag usage
+		flagSet.Usage = func() {}
+	default:
+		flagSet = flag.NewFlagSet(structName, flag.ExitOnError)
+		flagSet.Usage = func() {
+			fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+			flagSet.PrintDefaults()
+			fmt.Fprintf(os.Stderr, "\nGenerated environment variables:\n")
+			e := &EnvironmentLoader{
+				Prefix:    f.EnvPrefix,
+				CamelCase: f.CamelCase,
+			}
+			e.PrintEnvs(s)
+			fmt.Println("")
```

Comment:
i dont think this is required. If it is there for UI, better to use `fmt.Fprintf(os.Stderr, "\n")`